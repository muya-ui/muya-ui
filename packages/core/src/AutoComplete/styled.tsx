import styled, { css } from 'styled-components';

import ellipsisStyle from '../styled/mixins/ellipsis';

export const StyledAutoCompleteInputWrapper = styled.span`
  position: relative;
  display: inline-block;
  width: 100%;
`;

export const StyledPirmaryField = styled.span`
  ${ellipsisStyle};
`;

export const StyledSecondaryField = styled.span`
  flex-shrink: 0;
  margin-left: 12px;
`;

export const StyledAutoCompleteItem = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  ${props => css`
    ${StyledSecondaryField} {
      color: ${props.theme.colors.pattern.text.assistant};
    }
  `};
`;
