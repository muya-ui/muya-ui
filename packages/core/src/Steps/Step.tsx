import React, { Ref, useMemo } from 'react';

import { CloseIcon, SelectIcon } from '@muya-ui/theme-light';

import Typography from '../Typography';
import useStyles from '../utils/useStyles';
import useTheme from '../utils/useTheme';
import {
  StyledCircle,
  StyledDescriptionWrapper,
  StyledProgressLine,
  StyledStepContent,
  StyledStepWrapper,
  StyledTitle,
} from './styled';
import { IStepProps, IStepStyleKeys } from './types';
import { ICustomStylePropMap } from '../types';
import memoForwardRef from '../utils/memoForwardRef';

export default memoForwardRef((props: IStepProps, ref: Ref<HTMLDivElement>) => {
  const {
    onStepClick,
    stepNumber = 0,
    status,
    title,
    description,
    styles,
    size,
    icon,
    ...other
  } = props;
  const theme = useTheme();

  const {
    components: {
      Steps: { finishIcon: FinishIcon = SelectIcon, errorIcon: ErrorIcon = CloseIcon },
    },
  } = theme;
  const defaultStyles = useMemo<ICustomStylePropMap<IStepStyleKeys>>(
    () => ({
      content: '',
      circle: '',
      title: '',
      progressLine: '',
      descriptionWrapper: '',
      text: '',
    }),
    [],
  );
  const innerStyles = useStyles('step', defaultStyles, styles);

  let innerIcon: React.ReactNode;
  if (status === 'finish') {
    innerIcon = <FinishIcon />;
  } else if (status === 'error') {
    innerIcon = <ErrorIcon />;
  } else {
    innerIcon = stepNumber + 1;
  }

  if (icon) {
    innerIcon = icon;
  }

  const handleClick = useMemo(
    () =>
      onStepClick
        ? () => {
            onStepClick(stepNumber);
          }
        : undefined,
    [onStepClick, stepNumber],
  );

  return (
    <StyledStepWrapper ref={ref} theme={theme} {...other}>
      <StyledStepContent
        {...innerStyles.content}
        onClick={handleClick}
        theme={theme}
        status={status}
        size={size}
      >
        <StyledCircle {...innerStyles.circle} theme={theme} size={size} $bordered={!icon}>
          {innerIcon}
        </StyledCircle>
        <StyledTitle {...innerStyles.title} theme={theme}>
          {title}
        </StyledTitle>
        <StyledProgressLine {...innerStyles.progressLine} theme={theme} status={status} />
      </StyledStepContent>

      <StyledDescriptionWrapper {...innerStyles.descriptionWrapper} theme={theme} size={size}>
        <Typography.Text
          type={status === 'error' ? 'error' : undefined}
          {...innerStyles.text}
          color="assistant"
        >
          {description}
        </Typography.Text>
      </StyledDescriptionWrapper>
    </StyledStepWrapper>
  );
});
