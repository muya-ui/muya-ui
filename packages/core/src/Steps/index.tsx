export { default } from './Steps';
export { default as Steps } from './Steps';
export { default as Step } from './Step';
export * from './types';
