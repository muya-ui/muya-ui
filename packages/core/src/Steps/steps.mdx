---
name: Steps 步骤条
menu: 组件
route: /components/steps
autoDemos: true
parent: 导航
disableDoubleRow: true
---

# Steps 步骤条

引导用户按照流程完成任务的导航条。

## 使用场景

当任务复杂或者存在先后关系时，可以将其分解成一系列步骤，从而简化任务，避免操作内容过多给用户造成心理压力。

## 使用原则

- 需展示一共具有多少步骤，并表明当前所处步骤
- 请勿进行步骤条嵌套或在同个页面中使用多个步骤条

{Demos}

### API

#### IStepsProps

<Props propsName="IStepsProps" />

#### IStepProps

<Props propsName="IStepProps" />

### Styles

#### IStepStyleKeys

<Styles typeName="IStepStyleKeys" componentName="step" />
