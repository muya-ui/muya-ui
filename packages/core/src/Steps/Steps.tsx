import React, { Children, cloneElement, Ref, useCallback } from 'react';

import useTheme from '../utils/useTheme';
import { StyledStepsWrapper } from './styled';
import { IStepProps, IStepsProps } from './types';
import memoForwardRef from '../utils/memoForwardRef';

export default memoForwardRef((props: IStepsProps, ref: Ref<HTMLDivElement>) => {
  const { onChange, current = 0, children, status: currentStatus, size = 'm', ...other } = props;
  const theme = useTheme();

  const onStepClick = useCallback(
    (next: number) => {
      if (onChange && current !== next) {
        onChange(next);
      }
    },
    [current, onChange],
  );

  return (
    <StyledStepsWrapper theme={theme} ref={ref} {...other}>
      {Children.map(children, (child, index) => {
        // 默认等待中
        let status: IStepProps['status'] = 'wait';

        if (index === current) {
          // 进行中
          status = currentStatus || 'process';
        }

        if (index < current) {
          // 已完成
          status = 'finish';
        }

        const childProps: IStepProps = {
          status,
          stepNumber: index,
          ...child.props,
          size,
          onStepClick: onChange && onStepClick,
        };

        return cloneElement(child, childProps);
      })}
    </StyledStepsWrapper>
  );
});
