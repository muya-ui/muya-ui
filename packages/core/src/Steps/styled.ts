import styled, { css } from 'styled-components';

import { IThemedBaseProps } from '../types';
import { IStepProps } from './types';

interface IStepCircleProps extends IStepProps {
  $bordered?: boolean;
}

function makeStepContentStyle(props: IStepProps & IThemedBaseProps) {
  const { status, onClick, theme, size } = props;
  const { color, background, fontSize, lineHeight, fontWeight } = theme.components.Steps.content;
  const titleToken = theme.components.Steps.title;

  const titleColor = titleToken.color[status!];
  const finalColor = color[status!];
  const finalBgColor = background[status!];
  const hasBg = finalBgColor !== 'transparent' && finalBgColor !== theme.colors.spec.light;
  const colorStyle =
    status !== 'wait' &&
    css`
      color: ${finalColor};

      & ${StyledCircle} {
        border-color: ${hasBg ? finalBgColor : finalColor};
        background: ${finalBgColor};
      }
      & ${StyledTitle} {
        color: ${titleColor};
      }
    `;

  /**
   * 可点击状态
   * 1. 传入了onChange事件
   * 2. 当前不处于process状态
   * 3. 当前不处于error状态
   */
  const canClick = onClick && status !== 'process' && status !== 'error';

  const canClickStyle =
    canClick &&
    css`
      &:hover {
        color: ${color.hover};
        ${StyledCircle} {
          border-color: ${hasBg ? background.hover : color.hover};
          background: ${background.hover};
        }
        ${StyledTitle} {
          color: ${titleToken.color.hover};
        }
      }
      &:active {
        color: ${color.clicked};
        ${StyledCircle} {
          border-color: ${hasBg ? background.clicked : color.clicked};
          background: ${background.clicked};
        }
        ${StyledTitle} {
          color: ${titleToken.color.clicked};
        }
      }

      cursor: pointer;
    `;

  return css`
    display: flex;
    align-items: center;
    position: relative;
    color: ${color.wait};
    font-size: ${fontSize[size!]}px;
    line-height: ${lineHeight}px;
    font-weight: ${fontWeight};
    cursor: default;
    ${colorStyle}
    ${canClickStyle}
    &>* {
      ${props => {
        const {
          theme: { transition },
        } = props;
        return css`
          transition: all ${transition.pattern.easing.status}
            ${transition.pattern.duration.status}ms;
        `;
      }}
    }
  `;
}

export const StyledProgressLine = styled.div<IStepProps>`
  position: absolute;
  top: 50%;
  left: 100%;
  width: 9999px;
  height: 1px;
  transform: translateY(-50%);
  background-color: ${props => {
    const { status, theme } = props;
    const token = theme.components.Steps.progressLine;
    return status === 'finish' ? token.finishBgColor : token.bgColor;
  }};
`;

export const StyledStepWrapper = styled.div`
  display: flex;
  flex-flow: column nowrap;
  align-items: flex-start;
  overflow: hidden;
  flex: 1;
  margin-right: ${props => props.theme.spacing.spec.s4}px;

  &:last-child {
    flex: none;
    ${StyledProgressLine} {
      display: none;
    }
  }
`;

export const StyledCircle = styled.span<IStepCircleProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
  user-select: none;
  ${props => {
    const { theme, size, $bordered = true } = props;
    const borderStyle =
      $bordered &&
      css`
        border: 1px solid ${theme.colors.pattern.background.divider};
      `;
    const token = theme.components.Steps.circle;
    return css`
      margin-right: ${token.marginRight[size!]}px;
      width: ${token.size[size!]}px;
      height: ${token.size[size!]}px;
      ${borderStyle}
    `;
  }}
`;

export const StyledTitle = styled.span<IStepProps>`
  margin-right: ${props => props.theme.spacing.spec.s1}px;
`;

export const StyledStepContent = styled.div<IStepProps>`
  ${makeStepContentStyle}
`;

export const StyledDescriptionWrapper = styled.div<IStepProps>`
  ${props => {
    const token = props.theme.components.Steps.description;
    return css`
      margin-left: ${token.marginLeft[props.size!]}px;
      padding-left: ${token.paddingLeft[props.size!]}px;
      margin-top: ${token.marginTop}px;
      max-width: ${token.maxWidth}px;
      overflow: hidden;
    `;
  }}
`;

export const StyledStepsWrapper = styled.div`
  display: flex;
  width: 100%;
  margin: 0;
  padding: 0;
  font-size: 0;
`;
