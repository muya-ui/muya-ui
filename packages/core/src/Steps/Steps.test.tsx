import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import muyaThemeLight from '@muya-ui/theme-light';

import Step from './Step';
import Steps from './Steps';
import { StyledCircle, StyledStepContent } from './styled';

test('Steps should give children status and stepNumber', () => {
  const wrapper = mount(
    <Steps current={1}>
      <Step id="Step1" title="111" description="111" />
      <Step id="Step2" title="222" description="222" />
      <Step id="Step3" title="333" description="333" />
    </Steps>,
  );
  expect(
    wrapper
      .find('#Step1')
      .at(0)
      .prop('stepNumber'),
  ).toBe(0);
  expect(
    wrapper
      .find('#Step2')
      .at(0)
      .prop('stepNumber'),
  ).toBe(1);
  expect(
    wrapper
      .find('#Step3')
      .at(0)
      .prop('stepNumber'),
  ).toBe(2);

  expect(
    wrapper
      .find('#Step1')
      .at(0)
      .prop('status'),
  ).toBe('finish');
  expect(
    wrapper
      .find('#Step2')
      .at(0)
      .prop('status'),
  ).toBe('process');
  expect(
    wrapper
      .find('#Step3')
      .at(0)
      .prop('status'),
  ).toBe('wait');

  wrapper
    .find('#Step1')
    .at(0)
    .find(StyledStepContent)
    .at(0)
    .simulate('click');

  expect(
    wrapper
      .find('#Step1')
      .at(0)
      .prop('status'),
  ).toBe('finish');
});

test('should not emit onChange', () => {
  const handleChange = sinon.spy();

  const wrapper = mount(
    <Steps onChange={handleChange}>
      <Step id="Step1" title="111" description="111" />
      <Step id="Step2" title="222" description="222" />
      <Step id="Step3" title="333" description="333" />
    </Steps>,
  );

  wrapper
    .find('#Step1')
    .at(0)
    .find(StyledStepContent)
    .at(0)
    .simulate('click');

  expect(handleChange.called).toBe(false);
});

test('should emit onChange', () => {
  const handleChange = sinon.spy();

  const wrapper = mount(
    <Steps onChange={handleChange}>
      <Step id="Step1" title="111" description="111" />
      <Step id="Step2" title="222" description="222" />
      <Step id="Step3" title="333" description="333" />
    </Steps>,
  );

  wrapper
    .find('#Step2')
    .at(0)
    .find(StyledStepContent)
    .at(0)
    .simulate('click');

  expect(handleChange.calledOnce).toBe(true);
});

test('Step should emit onClick', () => {
  const handleChange = sinon.spy();
  const clickSpy = sinon.spy();

  const wrapper = mount(
    <Steps onChange={handleChange}>
      <Step id="Step1" onClick={clickSpy} title="111" description="111" />
      <Step id="Step2" title="222" description="222" />
      <Step id="Step3" title="333" description="333" />
    </Steps>,
  );

  wrapper
    .find('#Step1')
    .at(0)
    .find(StyledStepContent)
    .at(0)
    .simulate('click');

  expect(clickSpy.called).toBe(true);
});

test('Step should have default stepNumber', () => {
  const wrapper = mount(<Step status="wait" title="111" />);

  expect(wrapper.find(StyledCircle).getDOMNode().textContent).toBe('1');
});

test('Step should be error status', () => {
  const wrapper = mount(<Step status="error" title="111" />);

  expect(wrapper.find(StyledStepContent)).toHaveStyleRule(
    'color',
    muyaThemeLight.colors.spec.danger,
  );
});

test('should render different size', () => {
  const wrapper = renderer
    .create(
      <>
        <Step status="error" title="111" size="s" />
        <Step status="error" title="111" size="m" />
        <Step status="error" title="111" size="xl" />
        <Step status="error" title="111" size="l" />
      </>,
    )
    .toJSON();

  expect(wrapper).toMatchSnapshot();
});
