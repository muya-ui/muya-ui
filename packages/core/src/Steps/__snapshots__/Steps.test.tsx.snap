// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render different size 1`] = `
Array [
  .c4 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c10 {
  color: rgba(255,43,0,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c8 {
  position: absolute;
  top: 50%;
  left: 100%;
  width: 9999px;
  height: 1px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background-color: rgba(238,238,238,1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  overflow: hidden;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 8px;
}

.c0:last-child {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0:last-child .c7 {
  display: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-right: 4px;
  width: 24px;
  height: 24px;
  border: 1px solid rgba(238,238,238,1);
}

.c6 {
  margin-right: 2px;
}

.c11 .c2 {
  border-color: rgba(26,122,248,1);
  background: transparent;
}

.c11 .c5 {
  color: rgba(26,122,248,1);
}

.c12:hover .c2 {
  border-color: rgba(51,139,255,1);
  background: transparent;
}

.c12:hover .c5 {
  color: rgba(51,139,255,1);
}

.c12:active .c2 {
  border-color: rgba(12,108,235,1);
  background: transparent;
}

.c12:active .c5 {
  color: rgba(12,108,235,1);
}

.c13 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c13 .c5 {
  color: rgba(255,43,0,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgba(187,187,187,1);
  font-size: 12px;
  line-height: 16px;
  font-weight: 600;
  cursor: default;
  color: rgba(255,43,0,1);
}

.c1 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c1 .c5 {
  color: rgba(255,43,0,1);
}

.c1>* {
  -webkit-transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
  transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
}

.c14 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c14 .c5 {
  color: rgba(255,43,0,1);
}

.c15 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c15 .c5 {
  color: rgba(255,43,0,1);
}

.c9 {
  margin-left: 24px;
  padding-left: 4px;
  margin-top: 4px;
  max-width: 172px;
  overflow: hidden;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-step-content"
      size="s"
    >
      <span
        className="c2 c3 muya-step-circle"
        size="s"
      >
        <svg
          aria-hidden="false"
          className="c4"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z"
          />
        </svg>
      </span>
      <span
        className="c5 c6 muya-step-title"
      >
        111
      </span>
      <div
        className="c7 c8 muya-step-progress-line"
      />
    </div>
    <div
      className="c9 muya-step-description-wrapper"
      size="s"
    >
      <span
        className="c10 muya-step-text"
        color="assistant"
        type="error"
      />
    </div>
  </div>,
  .c4 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c10 {
  color: rgba(255,43,0,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c8 {
  position: absolute;
  top: 50%;
  left: 100%;
  width: 9999px;
  height: 1px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background-color: rgba(238,238,238,1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  overflow: hidden;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 8px;
}

.c0:last-child {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0:last-child .c7 {
  display: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-right: 6px;
  width: 28px;
  height: 28px;
  border: 1px solid rgba(238,238,238,1);
}

.c6 {
  margin-right: 2px;
}

.c11 .c2 {
  border-color: rgba(26,122,248,1);
  background: transparent;
}

.c11 .c5 {
  color: rgba(26,122,248,1);
}

.c12:hover .c2 {
  border-color: rgba(51,139,255,1);
  background: transparent;
}

.c12:hover .c5 {
  color: rgba(51,139,255,1);
}

.c12:active .c2 {
  border-color: rgba(12,108,235,1);
  background: transparent;
}

.c12:active .c5 {
  color: rgba(12,108,235,1);
}

.c13 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c13 .c5 {
  color: rgba(255,43,0,1);
}

.c14 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c14 .c5 {
  color: rgba(255,43,0,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgba(187,187,187,1);
  font-size: 14px;
  line-height: 16px;
  font-weight: 600;
  cursor: default;
  color: rgba(255,43,0,1);
}

.c1 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c1 .c5 {
  color: rgba(255,43,0,1);
}

.c1>* {
  -webkit-transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
  transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
}

.c15 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c15 .c5 {
  color: rgba(255,43,0,1);
}

.c9 {
  margin-left: 28px;
  padding-left: 6px;
  margin-top: 4px;
  max-width: 172px;
  overflow: hidden;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-step-content"
      size="m"
    >
      <span
        className="c2 c3 muya-step-circle"
        size="m"
      >
        <svg
          aria-hidden="false"
          className="c4"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z"
          />
        </svg>
      </span>
      <span
        className="c5 c6 muya-step-title"
      >
        111
      </span>
      <div
        className="c7 c8 muya-step-progress-line"
      />
    </div>
    <div
      className="c9 muya-step-description-wrapper"
      size="m"
    >
      <span
        className="c10 muya-step-text"
        color="assistant"
        type="error"
      />
    </div>
  </div>,
  .c4 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c10 {
  color: rgba(255,43,0,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c8 {
  position: absolute;
  top: 50%;
  left: 100%;
  width: 9999px;
  height: 1px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background-color: rgba(238,238,238,1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  overflow: hidden;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 8px;
}

.c0:last-child {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0:last-child .c7 {
  display: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-right: 8px;
  width: 40px;
  height: 40px;
  border: 1px solid rgba(238,238,238,1);
}

.c6 {
  margin-right: 2px;
}

.c11 .c2 {
  border-color: rgba(26,122,248,1);
  background: transparent;
}

.c11 .c5 {
  color: rgba(26,122,248,1);
}

.c12:hover .c2 {
  border-color: rgba(51,139,255,1);
  background: transparent;
}

.c12:hover .c5 {
  color: rgba(51,139,255,1);
}

.c12:active .c2 {
  border-color: rgba(12,108,235,1);
  background: transparent;
}

.c12:active .c5 {
  color: rgba(12,108,235,1);
}

.c13 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c13 .c5 {
  color: rgba(255,43,0,1);
}

.c14 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c14 .c5 {
  color: rgba(255,43,0,1);
}

.c15 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c15 .c5 {
  color: rgba(255,43,0,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgba(187,187,187,1);
  font-size: 16px;
  line-height: 16px;
  font-weight: 600;
  cursor: default;
  color: rgba(255,43,0,1);
}

.c1 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c1 .c5 {
  color: rgba(255,43,0,1);
}

.c1>* {
  -webkit-transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
  transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
}

.c9 {
  margin-left: 40px;
  padding-left: 8px;
  margin-top: 4px;
  max-width: 172px;
  overflow: hidden;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-step-content"
      size="xl"
    >
      <span
        className="c2 c3 muya-step-circle"
        size="xl"
      >
        <svg
          aria-hidden="false"
          className="c4"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z"
          />
        </svg>
      </span>
      <span
        className="c5 c6 muya-step-title"
      >
        111
      </span>
      <div
        className="c7 c8 muya-step-progress-line"
      />
    </div>
    <div
      className="c9 muya-step-description-wrapper"
      size="xl"
    >
      <span
        className="c10 muya-step-text"
        color="assistant"
        type="error"
      />
    </div>
  </div>,
  .c4 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c10 {
  color: rgba(255,43,0,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c8 {
  position: absolute;
  top: 50%;
  left: 100%;
  width: 9999px;
  height: 1px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background-color: rgba(238,238,238,1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  overflow: hidden;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 8px;
}

.c0:last-child {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0:last-child .c7 {
  display: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-right: 8px;
  width: 32px;
  height: 32px;
  border: 1px solid rgba(238,238,238,1);
}

.c6 {
  margin-right: 2px;
}

.c11 .c2 {
  border-color: rgba(26,122,248,1);
  background: transparent;
}

.c11 .c5 {
  color: rgba(26,122,248,1);
}

.c12:hover .c2 {
  border-color: rgba(51,139,255,1);
  background: transparent;
}

.c12:hover .c5 {
  color: rgba(51,139,255,1);
}

.c12:active .c2 {
  border-color: rgba(12,108,235,1);
  background: transparent;
}

.c12:active .c5 {
  color: rgba(12,108,235,1);
}

.c13 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c13 .c5 {
  color: rgba(255,43,0,1);
}

.c14 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c14 .c5 {
  color: rgba(255,43,0,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgba(187,187,187,1);
  font-size: 14px;
  line-height: 16px;
  font-weight: 600;
  cursor: default;
  color: rgba(255,43,0,1);
}

.c1 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c1 .c5 {
  color: rgba(255,43,0,1);
}

.c1>* {
  -webkit-transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
  transition: all cubic-bezier(0.4,0,0.2,1) 150ms;
}

.c15 .c2 {
  border-color: rgba(255,43,0,1);
  background: transparent;
}

.c15 .c5 {
  color: rgba(255,43,0,1);
}

.c9 {
  margin-left: 32px;
  padding-left: 8px;
  margin-top: 4px;
  max-width: 172px;
  overflow: hidden;
}

<div
    className="c0"
  >
    <div
      className="c1 muya-step-content"
      size="l"
    >
      <span
        className="c2 c3 muya-step-circle"
        size="l"
      >
        <svg
          aria-hidden="false"
          className="c4"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z"
          />
        </svg>
      </span>
      <span
        className="c5 c6 muya-step-title"
      >
        111
      </span>
      <div
        className="c7 c8 muya-step-progress-line"
      />
    </div>
    <div
      className="c9 muya-step-description-wrapper"
      size="l"
    >
      <span
        className="c10 muya-step-text"
        color="assistant"
        type="error"
      />
    </div>
  </div>,
]
`;
